### ISSUE TITLE
Add params to REST to fix SQL injection

### ISSUE THREAD

**Issue by amunra** [questdb/questdb#2025]
### Is your feature request related to a problem?

We've seen code like this in one of the tutorials:

```python
requests.get('http://localhost:9000/exec', params={
    'query': f'insert into excel_rates values({date}, {usd}, {inr})'})
```

String interpolation in the query exposes the customer (and database) to SQL injection.

### Describe the solution you'd like.

We should provide an additional way to support this via, for example, a `params` query parameter list and avoid this issue:

```python
# Note: This is just an idea to illustrate the point. Not the only solution.

query = f'insert into excel_rates values({date}, {usd}, {inr})'
requests.get('http://localhost:9000/exec', params={
    'query': 'insert into excel_rates values(?, ?, ?)',
    'params': [date, usd, inr]})
```

I.e. this would expand to, for example:

```
http://localhost:9000/exec?query=insert+into+excel_rates+values%28%3F%2C+%3F%2C+%3F%29&params=2022-08-06&params=10.4&params=1034.24
```

### Describe alternatives you've considered.

This can be worked around with client-side SQL encoding, but it's unrealistic and error-prone to expect our users to do this.

**Comment by puzpuzpuz** [comment]
I agree that this would be a nice addition. On the other hand, PG Wire should be considered a secure alternative to be used in user applications. HTTP API is only good for scripts and local interaction.

**Comment by HeZean** [comment]
Hi, I'd like to try this, can you assign it to me?

Up to now, I find these implemented classes may be helpful:

```
io.questdb.cutlass.text.types.TypeManager
io.questdb.griffin.engine.functions.bind.BindVariableServiceImpl
```

I think we can start by reusing these, like what `PGConnectionContext` does.

This may be close to the first solution mentioned in [#285 (comment)](https://github.com/questdb/questdb/issues/285#issuecomment-734346250).

**Comment by puzpuzpuz** [comment]
> Hi, I'd like to try this, can you assign it to me?

Sure, I've assigned you. Look forward to your contribution. 👍

As for #285, yes, that issue is certainly helpful.

**Comment by stekman** [comment]
Just another factor to think about: When using PG Wire I realized that for our use-case we need to send in table name and column names in a secure way to the prepared statement. This is not possible in JDBC. So what would be great addition here would be to have the possibility to use '?' for table name and columns as an addition to the possibilities in JDBC.

**Comment by bziobrowski** [comment]
@stekman  Could you explain the use case ? 
Prepared statements are normally optimized and planned once and then used multiple times . 
Tables and columns (I assume you don't mean string literals but sql identifiers) are essential to optimization so parameterizing them would force cached plan invalidation and then parsing/optimization/planning.

**Comment by stekman** [comment]
Yes, that is describing my problem and the need for this in the REST interface. Our application relies heavily on the CSV import feature and the column and table names come from this import. Thus we need to query the database with different table and column names each time, depending on file content and name of the table. So we will of course not get any optimization in planning etc, but it would help in protection vs sql injection  (which this feature request is about). MySql supports this via placeholders with a ?? notation in JDBC (not well documented and maybe not in all versions).

**Comment by HeZean** [comment]
This issue couldn't be fixed soon I'm afraid. Currently I'm preparing for my final exams -- I'll try be back on this in my summer vacation (in about 1mo), if anyone has brilliant idea and cannot wait to have it fixed, please pick it up.
